calc_decay.py

```diff
@@ -1,14 +1,47 @@
-from numpy import *
-import crpropa as crp
+import numpy as np
 from scipy.integrate import quad
 import gitHelp as gh
 import os
+from units import mass_electron, c_squared, c_light, h_planck, keV, amu
 
 cdir = os.path.split(__file__)[0]
 
 # Script to preprocess the nuclear decay data table from the BNL NuDat2 database
 # Decay Search: http://www.nndc.bnl.gov/nudat2/indx_sigma.jsp, output: formatted file --> decay_NuDat2.txt
 # Decay Radiation Search: gamma_NuDat2.txt: http://www.nndc.bnl.gov/nudat2/indx_dec.jsp --> gamma_NuDat2.txt
+#
+class NuclearMassTable(object):
+    """Class to provide tabulated nuclear masses
+    
+    This function reimplements the basic functionality of
+    CRPropa's particleMass module see here:
+    https://github.com/CRPropa/CRPropa3/blob/master/include/crpropa/ParticleMass.h
+    """
+
+    def __init__(self):
+        try:
+            self.massTable = np.loadtxt('data/nuclear_mass.txt', usecols=(2))
+        except FileNotFoundError:
+            print("The file 'data/nuclear_mass.txt' was not found.")
+            print("Run the script calc_mass.py and try again.")
+
+    def getMass(self, id: int) -> float:
+        """Helper function to return tabulated nuclear masses"""
```

Comment:
comment that this ID is not the typical PID in CRPropa.

Reply:
Done